teuthology/openstack/__init__.py

```diff
@@ -883,12 +883,14 @@ def create_security_group(self):
         among instances created within the same tenant.
         """
         try:
-            self.run("security group show teuthology")
+            self.run("security group show teuthology && security group show teuthology-worker")
             return
         except subprocess.CalledProcessError:
             pass
         # FIXME: the following assumes "--name teuthology"
```

Comment:
This code suits better for this purpose and there is no need to check process error

```
groups = misc.sh('openstack security group list -C Name -f value').split('\n')
if all(g in groups for g in ['teuthology', 'teuthology-worker']):
    return
```